src/shared/components/palette/document/read-document-field.component.ts

```diff
@@ -1,8 +1,29 @@
 import { Component } from '@angular/core';
 import { AbstractFieldReadComponent } from '../base-field/abstract-field-read.component';
+import { Router } from '@angular/router';
+import { WindowService } from '../../../services/window';
+import { DocumentManagementService } from '../../../services/document-management';
+
+const MEDIA_VIEWER = 'media-viewer';
 
 @Component({
   selector: 'ccd-read-document-field',
   templateUrl: './read-document-field.html'
 })
-export class ReadDocumentFieldComponent extends AbstractFieldReadComponent {}
+export class ReadDocumentFieldComponent extends AbstractFieldReadComponent {
+
+  constructor(private windowService: WindowService,
+              private documentManagement: DocumentManagementService,
+              private router: Router) {
+    super();
+  }
+
+  showMediaViewer(): void {
+    let routerMediaViewer = this.router.createUrlTree(['/media-viewer']);
+    this.windowService.removeLocalStorage(MEDIA_VIEWER);
+    if (this.caseField && this.caseField.value) {
+      this.windowService.setLocalStorage(MEDIA_VIEWER, this.documentManagement.getMediaViewerInfo(this.caseField.value));
```

Comment:
can we find a better name for the constant `MEDIA_VIEWER`? Perhaps `MEDIA_VIEWER_INFO` is a bit more clear?

Reply:
Done.